evalml/automl/automl_search.py

```diff
@@ -328,7 +328,7 @@ def _validate_data_checks(self, data_checks):
             return AutoMLDataChecks(data_checks)
```

Comment:
Although this PR is for the DefaultDataChecks, I spent a while trying to figure out how we might pass it to DataChecks or AutoMLDataChecks. It becomes a bigger issue in which we have to check and see if ClassImbalanceDataCheck is one of the Data Checks, then determine what parameters were passed by the user, and if `num_cv_folds` is one of them. Then overwriting that to reflect `self.data_splitter.get_n_splits()`. There's probably an easier way I'm sure :)

Reply:
I think it's fine to only make the change for `DefaultDataChecks`! If the user is providing their own data checks, it's assumed they init them with the appropriate parameters!